### ISSUE TITLE
Not connecting? 

### ISSUE THREAD

**Issue by cspiliakos** [VivaPayments/API#1255]
Recently updated WHMCS, and can't get through any payments. 

Here's the debug log:

06/Dec/2019:11:32:34 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:32:34 [Link] License: Active
06/Dec/2019:11:32:34 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:32:34 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:32:34 [Link] Table mod_vivapay checked
06/Dec/2019:11:32:34 [Link] Installment periods available: 10
06/Dec/2019:11:32:34 [Link] Sending form
06/Dec/2019:11:32:34 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:32:36 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:32:36 [Link] License: Active
06/Dec/2019:11:32:36 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:32:36 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:32:36 [Link] Table mod_vivapay checked
06/Dec/2019:11:32:36 [Link] Installment periods available: 10
06/Dec/2019:11:32:36 [Link] Sending form
06/Dec/2019:11:32:36 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:32:43 [Formredir] VivaPayments 1.14 - Start
06/Dec/2019:11:32:43 [Formredir] License: Active
06/Dec/2019:11:32:43 [Formredir] POST vars: transid=bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:32:43 [Formredir] State of use_installments cookie: false
06/Dec/2019:11:32:43 [Formredir] Executing query: SELECT * FROM `mod_vivapay` WHERE `transid` = 'bcde550acbcf062777e632da07b67e55'
06/Dec/2019:11:32:43 [Formredir] Communication method: cURL
06/Dec/2019:11:32:43 [Formredir] Postfields: Amount=1000, RequestLang=en-US, FullName=Konstantinos Spiliakos, Email=xxxx, Phone=xxxx, MaxInstallments=0, MerchantTrns=31, CustomerTrns=xxxx, SourceCode=WHMCS
06/Dec/2019:11:32:44 [Formredir] cURL reply: 
06/Dec/2019:11:32:44 [Formredir] Error (Result is not a json object) - Redirecting to http://cp.xtremeinfinity.com/cp//viewinvoice.php?id=31&paymentfailed=true
06/Dec/2019:11:32:44 [Formredir] VivaPayments 1.14 - End 
06/Dec/2019:11:32:46 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:32:46 [Link] License: Active
06/Dec/2019:11:32:46 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:32:46 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:32:46 [Link] Table mod_vivapay checked
06/Dec/2019:11:32:46 [Link] Installment periods available: 10
06/Dec/2019:11:32:46 [Link] Sending form
06/Dec/2019:11:32:46 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:33:02 [Formredir] VivaPayments 1.14 - Start
06/Dec/2019:11:33:02 [Formredir] License: Active
06/Dec/2019:11:33:02 [Formredir] POST vars: transid=bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:33:02 [Formredir] State of use_installments cookie: false
06/Dec/2019:11:33:02 [Formredir] Executing query: SELECT * FROM `mod_vivapay` WHERE `transid` = 'bcde550acbcf062777e632da07b67e55'
06/Dec/2019:11:33:02 [Formredir] Communication method: cURL
06/Dec/2019:11:33:02 [Formredir] Postfields: Amount=1000, RequestLang=en-US, FullName=Konstantinos Spiliakos, Email=xxxx, Phone=xxxx, MaxInstallments=0, MerchantTrns=31, CustomerTrns=xxxx, SourceCode=WHMCS
06/Dec/2019:11:33:02 [Formredir] cURL reply: 
06/Dec/2019:11:34:04 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:34:04 [Link] License: Active
06/Dec/2019:11:34:04 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:34:04 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:34:04 [Link] Table mod_vivapay checked
06/Dec/2019:11:34:04 [Link] Installment periods available: 10
06/Dec/2019:11:34:04 [Link] Sending form
06/Dec/2019:11:34:04 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:34:06 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:34:06 [Link] License: Active
06/Dec/2019:11:34:06 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:34:06 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:34:06 [Link] Table mod_vivapay checked
06/Dec/2019:11:34:06 [Link] Installment periods available: 10
06/Dec/2019:11:34:06 [Link] Sending form
06/Dec/2019:11:34:06 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:34:08 [Formredir] VivaPayments 1.14 - Start
06/Dec/2019:11:34:08 [Formredir] License: Active
06/Dec/2019:11:34:08 [Formredir] POST vars: transid=bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:34:08 [Formredir] State of use_installments cookie: false
06/Dec/2019:11:34:08 [Formredir] Executing query: SELECT * FROM `mod_vivapay` WHERE `transid` = 'bcde550acbcf062777e632da07b67e55'
06/Dec/2019:11:34:08 [Formredir] Communication method: fopen
06/Dec/2019:11:34:08 [Formredir] Postfields: Amount=1000, RequestLang=en-US, FullName=Konstantinos Spiliakos, Email=xxxx, Phone=xxxx, MaxInstallments=0, MerchantTrns=31, CustomerTrns=xxxx, SourceCode=WHMCS
06/Dec/2019:11:39:09 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:39:09 [Link] License: Active
06/Dec/2019:11:39:09 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:39:09 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:39:09 [Link] Table mod_vivapay checked
06/Dec/2019:11:39:09 [Link] Installment periods available: 10
06/Dec/2019:11:39:09 [Link] Sending form
06/Dec/2019:11:39:09 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:39:50 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:39:50 [Link] License: Active
06/Dec/2019:11:39:50 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:39:50 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:39:50 [Link] Table mod_vivapay checked
06/Dec/2019:11:39:50 [Link] Installment periods available: 10
06/Dec/2019:11:39:50 [Link] Sending form
06/Dec/2019:11:39:50 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:39:53 [Formredir] VivaPayments 1.14 - Start
06/Dec/2019:11:39:53 [Formredir] License: Active
06/Dec/2019:11:39:53 [Formredir] POST vars: transid=bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:39:53 [Formredir] State of use_installments cookie: false
06/Dec/2019:11:39:53 [Formredir] Executing query: SELECT * FROM `mod_vivapay` WHERE `transid` = 'bcde550acbcf062777e632da07b67e55'
06/Dec/2019:11:39:53 [Formredir] Communication method: cURL
06/Dec/2019:11:39:53 [Formredir] Postfields: Amount=1000, RequestLang=en-US, FullName=Konstantinos Spiliakos, Email=xxxx, Phone=xxxx, MaxInstallments=0, MerchantTrns=31, CustomerTrns=xxxx, SourceCode=WHMCS
06/Dec/2019:11:39:53 [Formredir] cURL reply: <html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>

06/Dec/2019:11:39:53 [Formredir] Error (Result is not a json object) - Redirecting to http://cp.xtremeinfinity.com/cp//viewinvoice.php?id=31&paymentfailed=true
06/Dec/2019:11:39:53 [Formredir] VivaPayments 1.14 - End 
06/Dec/2019:11:39:53 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:39:53 [Link] License: Active
06/Dec/2019:11:39:53 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:39:53 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:39:53 [Link] Table mod_vivapay checked
06/Dec/2019:11:39:53 [Link] Installment periods available: 10
06/Dec/2019:11:39:53 [Link] Sending form
06/Dec/2019:11:39:53 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:41:32 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:41:32 [Link] License: Active
06/Dec/2019:11:41:32 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:41:32 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:41:32 [Link] Table mod_vivapay checked
06/Dec/2019:11:41:32 [Link] Installment periods available: 10
06/Dec/2019:11:41:32 [Link] Sending form
06/Dec/2019:11:41:32 [Link] VivaPayments 1.14 - End
06/Dec/2019:11:41:35 [Formredir] VivaPayments 1.14 - Start
06/Dec/2019:11:41:35 [Formredir] License: Active
06/Dec/2019:11:41:35 [Formredir] POST vars: transid=bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:41:35 [Formredir] State of use_installments cookie: false
06/Dec/2019:11:41:35 [Formredir] Executing query: SELECT * FROM `mod_vivapay` WHERE `transid` = 'bcde550acbcf062777e632da07b67e55'
06/Dec/2019:11:41:35 [Formredir] Communication method: cURL
06/Dec/2019:11:41:35 [Formredir] Postfields: Amount=1000, RequestLang=en-US, FullName=Konstantinos Spiliakos, Email=xxxx, Phone=xxxx, MaxInstallments=0, MerchantTrns=31, CustomerTrns=xxxx, SourceCode=WHMCS
06/Dec/2019:11:41:35 [Formredir] cURL reply: 
06/Dec/2019:11:41:35 [Formredir] Error (Result is not a json object) - Redirecting to http://cp.xtremeinfinity.com/cp//viewinvoice.php?id=31&paymentfailed=true
06/Dec/2019:11:41:35 [Formredir] VivaPayments 1.14 - End 
06/Dec/2019:11:41:35 [Link] VivaPayments 1.14 - Start
06/Dec/2019:11:41:35 [Link] License: Active
06/Dec/2019:11:41:35 [Link] currency=EUR, amount=10.00, amountbase=10.00
06/Dec/2019:11:41:35 [Link] transid created: bcde550acbcf062777e632da07b67e55
06/Dec/2019:11:41:35 [Link] Table mod_vivapay checked
06/Dec/2019:11:41:35 [Link] Installment periods available: 10
06/Dec/2019:11:41:35 [Link] Sending form
06/Dec/2019:11:41:35 [Link] VivaPayments 1.14 - End

**Comment by cspiliakos** [comment]
Το περίεργο του πράγματος συνεχίζεται, διότι βγάζει off το allow_url_fopen, παρόλο που στο php.ini εμφανίζεται ενεργοποιημένο.

**Comment by matthewrgourd** [comment]
Hi @cspiliakos, sorry to hear this. Can you tell me a bit more about your implementation of Viva Wallet? Is it a custom integration through our [Payment API](https://developer.vivawallet.com/api-reference-guide/payment-api/) or have you used [Native Checkout v2](https://developer.vivawallet.com/online-checkouts/native-checkout-v2/)? Alternatively, have you installed one of our plugins on an [e-commerce platform](https://developer.vivawallet.com/e-commerce-plugins/)?

**Comment by cspiliakos** [comment]
> Hi @cspiliakos, sorry to hear this. Can you tell me a bit more about your implementation of Viva Wallet? Is it a custom integration through our [Payment API](https://developer.vivawallet.com/api-reference-guide/payment-api/) or have you used [Native Checkout v2](https://developer.vivawallet.com/online-checkouts/native-checkout-v2/)? Alternatively, have you installed one of our plugins on an [e-commerce platform](https://developer.vivawallet.com/e-commerce-plugins/)?

Hi @matthewrgourd, it's a plugin through a 3rd party, for WHMCS. Last time I checked, Viva has not released a WHMCS plugin yet. It's Native v2 (at least according to my panel). 

My question though, is if it's supposed to be working, and a similar one, but older was working before the update of WHMCS, and Plesk, why is it not working now? The 3rd party developer asked me to contact you, as they mention that I need to check with Viva Wallet's IT team to see if the requests are arriving properly, or if something is not going through right. 

And a bit off topic, but quite useful, in my mind github should not be an official support place for a company the size of Viva. General development questions yes, but questions regarding connectivity and/or specialized support should be directed to a dedicated support site for ticket assignment and tracking, or have the ability to open it through our merchant's panel. 

Thanks :)

Kind regards,
cspiliakos

**Comment by matthewrgourd** [comment]
Hi @cspiliakos , I'm afraid the fact that you're using a third-party plugin means we're unable to provide support. If it was working before but isn't now may have something to do with the new 3DS component in our Native Checkout. Beyond that, I can't provide further help unfortunately. In reply to your other query, we intend to have a fully operational technical support department in the new year. Thanks and sorry I can't be of more help.

**Comment by cspiliakos** [comment]
Hi @matthewrgourd, the Issue was resolved earlier last night by the dev. of the plugin. Apparently the password given by Viva was messing up with the plugin (weird, but ok, it can happen). The character " > " that was in the password was messing up the plugin, and viva's servers did not reply that there was an issue with the login.

On a more personal level, I'm not happy by Viva's support. Your phone support told me to post here, and I've mentioned multiple times to them that I need to talk to someone on the phone for obvious reasons, as it's not something resolved over a GitHub thread. You've closed a thread with the reason of not using an official plugin by Viva, ok, even though a client was having a problem using your services, and there is no official package or plugin by Viva!!! I'm really afraid to ask what would happen if a client had more serious issues with you guys. 

In any case, I've been happy most of the times I've had to use Viva as a merchant to receive goods, but you guys need to work on your support skills.
